### PR TITLE
Enable selecting whether new entities are enabled by default in Config Flow

### LINKED ISSUE #27: Security (Create alerts for unknown devices connected to your wifi)
How do I set this up?  What trigger would I use to send a notification when an unknown device connects to my Wifi?

### PR DESCRIPTION
**Pull request by hawksj**
This is the largest change I've made to date. It creates resolves #27 which was initially solved with PR #31 but was later reverted in PR #33 due to #32. That has been fixed in this PR and I have re-added the integration many times with the tickbox checked and unchecked and I have had no log errors.

This change adds a new tickbox during the config flow which allows the user to select whether they would like new entities to be automatically enabled when discovered. I added this as personally, I prefer entities to be automatically enabled but enabling it by default for all instances and not being able to change it without modifying the integration code isn't ideal for many users and could cause the database to grow too large in some installations as stated in https://developers.home-assistant.io/docs/core/entity/#generic-properties.
image

This PR also makes minor changes to the translations files to make the config flow look nicer and describe the fields better.

I had to learn a lot to make this work and I think it looks fine but I would appreciate someone with more experience to give me the thumbs up that it looks okay.

### CODE REVIEW

**Comment by hawksj** [comment]
I've posted this as a draft because I want to go through and rename the key to something less vague than "enable". This was a temporary value while I was testing. I will open this PR for review once that has been changed.

**Comment by hawksj** [comment]
Keys have been renamed to make them less vague. 

@atudor2 sorry to ask again but as this is the first time I've changed config entries, would you mind giving it a look over and a test to check that everything looks OK before I merge this please? Thank you.

**Comment by atudor2** [comment]
Hi @hawksj sure happy to give it a spin my side but I'm away from my main and dev HA instances for the next few weeks so will only be able to give feedback around the week of May 13 - happy to wait until then?

**Comment by hawksj** [comment]
Yes of course, thanks @atudor2. I'm not confident enough to merge without a second opinion so happy to wait a few weeks to make sure it's stable. Your help is appreciated as always :)

File: custom_components/amplifi/device_tracker.py
Comment (atudor2) [inline custom_components/amplifi/device_tracker.py]: As part of testing an upgrade of an existing install in my test instance, this threw a KeyError.
I suggest maybe doing a get() to handle cases where the new config item doesn't exist yet:

`if self.config_entry.data.get(CONF_ENABLE_NEW_DEVICES, False):`

**Review by atudor2 — CHANGES_REQUESTED** [review]
Hi @hawksj I installed the change into my dev instance and there is a small bug for upgrades to an existing config entry - once I patched it locally my side with a get() call it worked without issue

**Comment by hawksj** [comment]
Great catch, thanks a lot @atudor2. I'll make that change and test on my end too. I mostly stumbled through this by finding snippets and copying existing code so if that's the only bug then I'm pretty pleased with it! Thanks again. Sam

**Comment by hawksj** [comment]
The change has been made as Alistair recommended and I have no issues my end.

@atudor2 do you think it's appropriate to increment the minor version with this change? Perhaps 2.1.0 given that these are meaningful changes that users might want to respond to? I am also keen to create an actual release for the changes in this PR rather than showing users the commit hash when HACS prompts to update, and to better explain these changes.

_Unrelated but there is an issue with unloading the config entry, a series of calls results in calling a function which doesn't exist which throws an error. Maybe Puttyman was planning to add this at a later date and never got round to it? Off topic but it came up in my testing. I have created #49 for this._

**Comment by atudor2** [comment]
@hawksj Yep makes sense to bump the version and create a release imo 👍 

Re #49 I've seen this too but I'm also not very familiar with the HA Custom Component API so its hard to determine what needs to change to fix it. I have wondered if this is maybe due to a change on the HA side? I saw some rumblings on a dev blog (will put in a comment on the issue) and mess about in my test instance

**Review by atudor2 — APPROVED** [review]

**Comment by hawksj** [comment]
Thanks again Alistair. I think this is ready to merge now.

**Comment by atudor2** [comment]
:+1: - did one last spin in my test container and no issues

:shipit:

**Comment by hawksj** [comment]
Perfect thanks for the nod of approval! I will merge when I have the time to finish the release notes. Thanks for the help again!

**Pull request merged**
